Find a functionally equivalent code:android.widget.Switch.hitThumb:COMMENT
Method Modifier: private     
Comment:/**
 * @return true if (x, y) is within the target area of the switch thumb
 */

@@ -1,7 +1,9 @@
 {
+    // Relies on mTempRect, MUST be called first!
+    final int thumbOffset = getThumbOffset();
     mThumbDrawable.getPadding(mTempRect);
     final int thumbTop = mSwitchTop - mTouchSlop;
-    final int thumbLeft = mSwitchLeft + (int) (mThumbPosition + 0.5f) - mTouchSlop;
+    final int thumbLeft = mSwitchLeft + thumbOffset - mTouchSlop;
     final int thumbRight = thumbLeft + mThumbWidth + mTempRect.left + mTempRect.right + mTouchSlop;
     final int thumbBottom = mSwitchBottom + mTouchSlop;
     return x > thumbLeft && x < thumbRight && y > thumbTop && y < thumbBottom;

